### ISSUE TITLE
Need access to --filesystem=/usr/include 

### ISSUE THREAD

**Issue by Waqar144** [flathub/org.kde.kate#27]
When writing code, one often needs to access the header files in that directory. This is super important as if one can't jump to the location of header files the editor will not be usable.

Seems like just adding `--filesystem=/usr/include` will not work. Is this even supported?

`/usr/include` is just one example. There can be many such directories in the system that need to be accessible.

**Comment by Waqar144** [comment]
Not sure I understand you. Kate doesn't have a container plugin of any kind right now. And even if it did, how would it help?

Many users don't use containers for development. Missing access to /usr/ breaks workflow for myself in a pretty significant way.

**Comment by travier** [comment]
We could use `host-os` from https://man.archlinux.org/man/flatpak-metadata.5.en but it will need special handling anyway.

**Comment by Waqar144** [comment]
I think there is no way to solve this cleanly atm. Once you have access to /usr/, the difference between a sandbox / host-app is nil. Maybe we'll have a clean way to do this in future or maybe there already is. Will check what gnome-builder does as it also supports LSP and C/C++.

Regardless, the current state of flatpak-Kate is useful as-is for many users. A lot of new languages don't install their stuff in /usr, unless you really want. E.g., Go, dart, javascript and even rust can be in your home directory and will be accessible.

**Comment by travier** [comment]
GNOME Builder has support for running things in containers via podman or toolbox.

**Comment by KhazAkar** [comment]
Yes, but this is sort-of remote development support, which Kate does not currently have in any way, shape or form. Additionally, GNOME Builder seems much more as IDE than Kate, which is more like VSCode competitor, not Visual Studio (non VSCode one) one.

**Comment by christoph-cullmann** [comment]
I think the point is that Kate allows to use the tools from outside the container but we end up opening files inside of it if these output paths into /usr/....
For our use case having access to /usr/include of the host would be the right solution.
